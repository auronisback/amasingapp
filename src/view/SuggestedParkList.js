import React, {Component} from 'react';
import {ParkMarker} from "./ParkMarker";

/**
 * Component which manages the list of suggested parks.
 * */
export default class SuggestedParkList extends Component {
  /**
   * Renders the suggested park list.
   * */
  render() {
    return (
        <div>{this._buildSuggestedParks()}</div>
    );
  };

  /**
   * Builds the park lot markers.
   * @return ParkMarker[] The created park markers
   * */
  _buildSuggestedParks = () => {
    let markers = [];
    this.props.parks.forEach((elem, index) => {
      markers.push(
          <ParkMarker
              key={elem.id}
              parking={elem}
              label={index + 1}
          />
      );
    });
    return markers;
  };
}