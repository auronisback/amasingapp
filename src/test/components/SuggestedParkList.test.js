import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SuggestedParkList from "../../components/SuggestedParkList";

Enzyme.configure({adapter: new Adapter()});

describe('<SuggestedParkList>', () => {
  it('Should be empty when no suggested park has been given', () => {
    const list = shallow(<SuggestedParkList/>);
    expect(list).toMatchSnapshot();
  });

  it('Should show park lots when given', () => {
    const stubParks = [
      {
        id: 1,
        occupancy: 0.12,
        distance: 100,
        parkNumber: 6,
        streetType: 'primary',
        haveMes: false,
        start: {lat: 1, lng: 1},
        end: {lat: 2, lng: 2},
        label: 1
      }, {
        id: 2,
        occupancy: 0.22,
        distance: 105,
        parkNumber: 14,
        streetType: 'secondary',
        haveMes: true,
        start: {lat: 10, lng: 10},
        end: {lat: 20, lng: 20},
        label: 2
      }
    ];
    const list = shallow(<SuggestedParkList parks={stubParks}/>);
    expect(list).toMatchSnapshot();
  });
});