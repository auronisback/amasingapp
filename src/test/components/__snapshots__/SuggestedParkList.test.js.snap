// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuggestedParkList> Should be empty when no suggested park has been given 1`] = `<div />`;

exports[`<SuggestedParkList> Should show park lots when given 1`] = `
<div>
  <ParkMarker
    key="1"
    label={1}
    parking={
      Object {
        "distance": 100,
        "end": Object {
          "lat": 2,
          "lng": 2,
        },
        "haveMes": false,
        "id": 1,
        "label": 1,
        "occupancy": 0.12,
        "parkNumber": 6,
        "start": Object {
          "lat": 1,
          "lng": 1,
        },
        "streetType": "primary",
      }
    }
  />
  <ParkMarker
    key="2"
    label={2}
    parking={
      Object {
        "distance": 105,
        "end": Object {
          "lat": 20,
          "lng": 20,
        },
        "haveMes": true,
        "id": 2,
        "label": 2,
        "occupancy": 0.22,
        "parkNumber": 14,
        "start": Object {
          "lat": 10,
          "lng": 10,
        },
        "streetType": "secondary",
      }
    }
  />
</div>
`;
